src/desktop.c

```diff
@@ -77,16 +77,19 @@ desktop_focus_view(struct view *view, bool raise)
 		workspaces_switch_to(view->workspace, /*update_focus*/ false);
 	}
 
-	/*
-	 * Give input focus, even if the view claims not to want it (see
-	 * view->impl->wants_focus). This is a workaround for so-called
-	 * "globally active" X11 views (MATLAB known to be one such)
-	 * that expect to be able to control focus themselves, but can't
-	 * under labwc since it's disallowed at the wlroots level.
-	 */
-	struct seat *seat = &view->server->seat;
-	if (view->surface != seat->seat->keyboard_state.focused_surface) {
-		seat_focus_surface(seat, view->surface);
+	switch (view_wants_focus(view)) {
+	case VIEW_WANTS_FOCUS_ALWAYS:
+		; /* works around "a label can only be part of a statement" */
+		struct seat *seat = &view->server->seat;
+		if (view->surface != seat->seat->keyboard_state.focused_surface) {
+			seat_focus_surface(seat, view->surface);
+		}
+		break;
```

Comment:
nit: we can instead wrap them with `{ }` like this:

```suggestion
	case VIEW_WANTS_FOCUS_ALWAYS: {
		struct seat *seat = &view->server->seat;
		if (view->surface != seat->seat->keyboard_state.focused_surface) {
			seat_focus_surface(seat, view->surface);
		}
		break;
	}
```

Reply:
This was discussed before but the `; /* works around ..` has the benefit that it can be removed project wide relatively trivially once we deem it fine to not care of those old gcc versions anymore.

Reply:
Okay, then it's alright.